sql/core/src/main/scala/org/apache/spark/sql/execution/joins/LeftSemiJoinHash.scala

```diff
@@ -33,7 +33,8 @@ case class LeftSemiJoinHash(
     leftKeys: Seq[Expression],
     rightKeys: Seq[Expression],
     left: SparkPlan,
-    right: SparkPlan) extends BinaryNode with HashJoin {
+    right: SparkPlan,
+    condition: Option[Expression]) extends BinaryNode with HashJoin {
```

Comment:
The code here looks pretty much duplicated with the broadcast case.  Can we abstract the similar code in to a `SemiJoin` trait, similar to what we do in `HashJoin`?